### PR TITLE
Study sessions

### PR DESCRIPTION
**Pull request by mkaburis**
- [ ] Added create session activity
- [ ] Added session recycler view with custom layout
- [ ] Fixed a variety of Firebase Firestore issues

### CODE REVIEW

File: app/src/main/java/study_dev/testbed/studyhopper/models/Session.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
+package study_dev.testbed.studyhopper.models;
+
+import com.google.firebase.Timestamp;
+
+public class Session {
+    private String sessionName;
+    private String sessionDescription;
+    private String sessionLocation;
+    private Timestamp sessionDate;
+    private Timestamp sessionStartTime;
```
Comment (master-faz) [inline app/src/main/java/study_dev/testbed/studyhopper/models/Session.java:10]: what is the benefit of using timestamp over date?

File: app/src/main/java/study_dev/testbed/studyhopper/ui/dashboard/CreateGroup.java [hunk 2]
```diff
@@ -201,6 +200,11 @@ else if(groupPreference.equals("Males Only Group"))
 //        groupRef.add(groupTemplate);
 
         groupId = db.collection("groups").document().getId();
+
+        groupTemplate = new Group(groupName, courseCode, groupColor, preferenceSelected,
+                getUserName(), groupSizeMax, db.collection("groups").document(groupId), userUniversity,
```
Comment (master-faz) [inline app/src/main/java/study_dev/testbed/studyhopper/ui/dashboard/CreateGroup.java]: `db.collection("groups").document(groupId)`

I would store this in a variable. It looks messy and harder to read here

File: app/src/main/java/study_dev/testbed/studyhopper/ui/dashboard/DashboardFragment.java [hunk 3]
```diff
@@ -113,11 +113,14 @@ public void onComplete(@NonNull Task<QuerySnapshot> task) {
...
                                     public void onItemClick(DocumentSnapshot documentSnapshot, int position) {
                                         String id = documentSnapshot.getId();
+                                        Group mockGroup = documentSnapshot.toObject(Group.class);
+                                        String groupDocId = mockGroup.getDocumentId().getId();
                                         Toast.makeText(getContext(),
                                                 "Position: " + position + " ID: " + id, Toast.LENGTH_SHORT).show();
                                         Activity activity = (Activity) getContext();
                                         Intent intent = new Intent(getContext(), StudyGroupActivity.class);
                                         intent.putExtra("documentID", id);
+                                        intent.putExtra("groupDocId", groupDocId);
```
Comment (master-faz) [inline app/src/main/java/study_dev/testbed/studyhopper/ui/dashboard/DashboardFragment.java:121]: I would rename these to make it clearer which is which.

File: app/src/main/java/study_dev/testbed/studyhopper/ui/sessions/CreateSession.java [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+        sessionDescriptionEditText = findViewById(R.id.edit_text_session_description);
+        sessionLocationEditText = findViewById(R.id.edit_text_session_location);
+        sessionDateEditText = findViewById(R.id.edit_text_session_date);
+        sessionTimeStartEditText = findViewById(R.id.edit_text_session_start_time);
+        sessionTimeEndEditText = findViewById(R.id.edit_text_session_end_time);
+
+    }
+
+
+    public void selectSessionDate(View v) {
```
Comment (master-faz) [inline app/src/main/java/study_dev/testbed/studyhopper/ui/sessions/CreateSession.java:79]: Maybe would should abstract this into it's own class so we can use the same one for all calendars. The only thing is we would have to pass in the text boxes somehow.

**Review by master-faz — APPROVED** [review]

File: app/src/main/java/study_dev/testbed/studyhopper/models/Session.java (hunk 1 above)
Comment (mkaburis) [reply app/src/main/java/study_dev/testbed/studyhopper/models/Session.java:10]: Timestamp is utilized for Firebase. I retrieve the value from the Firestore as a Timestamp and then I convert it to a date. I could use a string if you think that is better.

File: app/src/main/java/study_dev/testbed/studyhopper/ui/dashboard/CreateGroup.java (hunk 2 above)
Comment (mkaburis) [reply app/src/main/java/study_dev/testbed/studyhopper/ui/dashboard/CreateGroup.java]: Resolved with variable for readability.

File: app/src/main/java/study_dev/testbed/studyhopper/ui/dashboard/DashboardFragment.java (hunk 3 above)
Comment (mkaburis) [reply app/src/main/java/study_dev/testbed/studyhopper/ui/dashboard/DashboardFragment.java:121]: Adjusted names to make it clearer.

File: app/src/main/java/study_dev/testbed/studyhopper/ui/sessions/CreateSession.java (hunk 4 above)
Comment (mkaburis) [reply app/src/main/java/study_dev/testbed/studyhopper/ui/sessions/CreateSession.java:79]: Yeah that's not a bad idea the dialogues are very useful for multiple areas of the app.

**Pull request merged**